clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -182,31 +182,32 @@ public void done(long baseOffset, long logAppendTime, RuntimeException exception
         Iterator<MutableRecordBatch> recordBatchIter = memoryRecords.batches().iterator();
         if (!recordBatchIter.hasNext())
             throw new IllegalStateException("Cannot split an empty producer batch.");
-        RecordBatch recordBatch = recordBatchIter.next();
-        if (recordBatchIter.hasNext())
-            throw new IllegalStateException("A producer batch should only have one record batch.");
-
         Iterator<Thunk> thunkIter = thunks.iterator();
         // We always allocate batch size because we are already splitting a big batch.
         // And we also Retain the create time of the original batch.
-        ProducerBatch batch = null;
-        for (Record record : recordBatch) {
-            assert thunkIter.hasNext();
-            Thunk thunk = thunkIter.next();
-            if (batch == null) {
-                batch = createBatchOffAccumulatorForRecord(record, splitBatchSize);
+        while (recordBatchIter.hasNext()) {
```

Comment:
Not sure I understand this logic: we expect `memoryRecords.batches()` to only have one batch but here we are expecting it to have many?